subsys/bluetooth/mesh/sensor_cli.c

```diff
@@ -141,17 +142,17 @@ static void handle_status(struct bt_mesh_model *mod,
 
 		sensor_status_id_decode(buf, &length, &id);
 		if (length == 0) {
-			if (is_rsp && rsp->id == id) {
-				rsp->value = NULL;
-				rsp->id = BT_MESH_PROP_ID_PROHIBITED;
-				model_ack_rx(&cli->ack);
-			}
-
+			BT_DBG("Failed to decode");
```

Comment:
I think this might be the wrong solution to this bug. If the format is 0x01, and the length field is 0x7f, the length should be interpreted as 0. In this case, the old code here was correct, and the bug is actually in `sensor_status_id_decode`, which doesn't apply the bitmask to the length to make it wrap around to 0.